### PR TITLE
hyperopt for MACDStrategy

### PR DESCRIPTION
**Pull request by hroff-1902**
Here it is, as it was proposed in #33, an implementation (as a beginning and inspiration for other authors) of the hyperopt for MACDStrategy.

Created from the generic hyperopt template, it allows to optimize the CCI band boundaries in the MACDStrategy strategy.

Thanks to @Prossi79, this is partly his work.

### CODE REVIEW

File: user_data/hyperopts/MACDStrategy_hyperopt.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+class MACDStrategy_hyperopt(IHyperOpt):
+    """
+    This is a test hyperopt to inspire you.
+    More information in https://github.com/freqtrade/freqtrade/blob/develop/docs/hyperopt.md
+     You can:
+    - Rename the class name (Do not forget to update class_name)
+    - Add any methods you want to build your hyperopt
+    - Add any lib you need to build your hyperopt
+     You must keep:
+    - the prototype for the methods: populate_indicators, indicator_space, buy_strategy_generator,
```
Comment (xmatthias) [inline user_data/hyperopts/MACDStrategy_hyperopt.py]: I don't wanna sound picky - but i think we should replace most of this doc-string with an explanation on what is optimized (this hyperopt file assumes you know you wanna use MACD crossover with CCI protection) - and you only optimize CCI. That's a valid usecase, and it's explained in the PR description - but that's not visible to the regular user ...

Example (from your comment):

> it allows to optimize the CCI band boundaries in the MACDStrategy strategy.

Comment (hroff-1902) [reply user_data/hyperopts/MACDStrategy_hyperopt.py]: agree, will change the doc strings

Comment (xmatthias) [reply user_data/hyperopts/MACDStrategy_hyperopt.py]: i guess you forgot about this just as i did ... ?

Comment (hroff-1902) [reply user_data/hyperopts/MACDStrategy_hyperopt.py]: A bit... ))

**Pull request merged**